sklearn/utils/validation.py

```diff
@@ -780,8 +780,9 @@ def check_array(
     elif hasattr(array, "iloc") and hasattr(array, "dtype"):
         # array is a pandas series
         pandas_requires_conversion = _pandas_dtype_needs_early_conversion(array.dtype)
-        if pandas_requires_conversion:
-            # Set to None, to convert to a np.dtype that works with array.dtype
+        if isinstance(array.dtype, np.dtype):
+            dtype_orig = np.result_type(array.dtype)
```

Comment:
If is already a np.dtype, we can use it directly?

```suggestion
            dtype_orig = array.dtype
```

Reply:
Of course. Was too much in "copy from the DF case" mode